### PR TITLE
[PDI-18508] changed toInt, toLong, and toDouble to match the new logi…

### PR DESCRIPTION
**Pull request by MarcDiamantHitachi**
…c for checking if string is null

### CODE REVIEW

**Comment by ddiroma** [comment]
@MarcDiamantHitachi Please add to the existing unit test cases for these methods to include at least tests for null values

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 11m 27s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 15m 26s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 10m 25s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 11m 36s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
core,plugins/streaming/impls/jms
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/di/core/Const.java [hunk 1]
```diff
@@ -1717,10 +1717,14 @@ public static long round( long f, int places, int roundingMode ) {
    */
   public static int toInt( String str, int def ) {
     int retval;
-    try {
-      retval = Integer.parseInt( str );
-    } catch ( Exception e ) {
+    if ( str == null ) {
       retval = def;
```
Comment (NJtwentyone) [inline core/src/main/java/org/pentaho/di/core/Const.java:1721]: is this just copying Apache's common lang NumberUtils function ?

 https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/math/NumberUtils.html#toInt-java.lang.String-int-
